File: Assignment_18/program5.c
// Accept N numbers from user and display all such elements which are multiples of 11.

///////////////////////////////////////////////////////////////////////////////
// File Name   :  program5.c
// Description :  Accept N numbers from user and display all such elements which are multiples of 11.
// Autor       :  Ankita Anil Patil
// Date        :  31/05/2025
///////////////////////////////////////////////////////////////////////////////

/*
 Input : N : 6
         Elements : 85 66 3 55 93 88
 Output: 66 55 88
 */

 #include<stdio.h>
 #include<stdlib.h>

 void Display(int Arr[] , int iSize)
 {
    if((Arr == NULL) || (iSize <= 0 ))
    {
        printf("Invalid input ");
        return FALSE;
    }
    
    int iCnt = 0;

    printf("Elements which are multiples of 11 are :\n");
    for(iCnt = 0; iCnt < iSize; iCnt++)
    {
        if((Arr[iCnt] % 11 == 0))
        {
            printf("%d\n",Arr[iCnt]);
        }
    }


 }

 int main()
 {

    int iLength = 0, iCnt = 0;
    int *ptr = NULL;

    printf("Enter number of element :\n");
    scanf("%d",&iLength);

    ptr = (int *) malloc(iLength * sizeof(int));

    if(NULL == ptr)
    {
        printf("Unable to allocate memory ");
        return  -1;
    }

    printf("Enter elements of array\n");

    for(iCnt = 0; iCnt < iLength; iCnt++)
    {
        scanf("%d",&ptr[iCnt]);
    }

    Display(ptr,iLength);

    free(ptr);
    ptr = NULL;
    
    return 0;
 }

File: Assignment_45/program5.java
//Write a program which accept matrix and check whether the matrix is Sparse matrix or not. 
// Sparse matrix is a matrix with the majority of its elements equal to zero.
/* 
Input : 
        1 0 3 0
        0 6 0 0
        0 0 1 0
        9 0 0 9

 Output : True 
 */

 
///////////////////////////////////////////////////////////////////////////////
// File Name   :  program5.java
// Description :  Accept matrix and check whether the matrix is Sparse matrix or not.
// Autor       :  Ankita Anil Patil
// Date        :  2/08/2025
///////////////////////////////////////////////////////////////////////////////


import java.util.*;
class Matrix
{
    private  int Arr[][];

    public Matrix(int A, int B)
    {
        Arr = new int[A][B];
        
    }
    
    public void Accept()
    {
        System.out.println("Please enter elements of matrix ");
      
        Scanner sobj = new Scanner(System.in);

        int i = 0, j = 0;

        for(i = 0; i < Arr.length; i++)
        {
            for(j = 0 ; j < Arr[i].length; j++)
            {
                Arr[i][j] = sobj.nextInt();
            }
            
            System.out.println();
        }
     
    }

     public void Display()
     {
         System.out.println("Elemnets of Matrix are : ");
         int i = 0, j = 0;
         
         for(i = 0; i < Arr.length; i++)
          {
               for(j = 0 ; j < Arr[i].length; j++)
               {
                     System.out.print(Arr[i][j]+"\t");
               }
               System.out.println();
            
          }
     }


    public boolean ChkSparse() 
    { 
          
          int i = 0, j = 0, iCount = 0, Total;
          Total = Arr.length * Arr[i].length;
          
          for(i = 0; i < Arr.length  ; i++)
          { 
                  for(j = 0 ; j < Arr[i].length; j++)
                  {
                        if(Arr[i][j] == 0)
                        {
                                iCount++;
                        }
                  }

               
            }

        if(iCount > Total /2)
        {
                return true;
        }
        else
        {
                return false;
        }
           
    }
 
   
}
     



class program5
{
    public static void main(String A[]) 
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter Number of rows ");
        int iRow = sobj.nextInt();

        System.out.println("Enter Number of columns ");
        int iCol = sobj .nextInt();

        Matrix mobj = new Matrix(iRow,iCol);
        mobj.Accept();
        mobj.Display();
        boolean bRet = mobj.ChkSparse();


        if(bRet == true)
        {
            System.out.println("True It is a Sparse Matrix");
        }
        else
        {
            System.out.println("Flase It is not a Sparse Matrix");
        }

        

    }
}

